File: src/DesignPattern/Behavioral/Observer/Observer-go/observer_Test.go
package main

func main() {
	var subject = &Subuject{}

	var obs = &ConcreteObserver{
		subject: subject,
	}

	subject.Attact(obs)

	for i := 0; i < 10; i++ {
		subject.SetState(i)
	}
}

File: src/DesignPattern/Behavioral/Observer/Observer-ts/observer.test.ts
import { ConcreteObserver, Subject } from "./observer";

let subject = new Subject();
// 创建观察者
let observer = new ConcreteObserver(subject);
// 添加观察者
subject.Attach(observer);

for (let i: number = 0; i < 10; i++) {
    subject.SetState(i);
}
for (let i: number = 5; i > 0; i--) {
    subject.SetState(i);
}